impeller/renderer/backend/gles/device_buffer_gles.h

```diff
@@ -33,8 +39,9 @@ class DeviceBufferGLES final
   ReactorGLES::Ref reactor_;
   HandleGLES handle_;
   std::string label_;
-  mutable std::shared_ptr<fml::Mapping> data_;
-  mutable bool uploaded_ = false;
+  mutable std::shared_ptr<Allocation> buffer_;
+  mutable bool dirty_ = false;
```

Comment:
IMO, the `dirty_` and `has_uploaded_` is becoming hard to reason about. The only reason `has_uploaded_` exists is because debug object labels cannot be set be on textures and buffers that are generated but not bound (they are so called "create on bind" resources). Unfortunately, `glCreateBuffers` and friends are not available on OpenGL ES 2. So the `has_uploaded_` field was tracking the create (via bind and data upload).

Perhaps you can add a field here that bumps up the generation of data whenever we get a `SetContents` call and another field that notes the generation that's been uploaded (so, `generation_` and `uploaded_generation_`). We can set debug labels and such when `uploaded_generation > 0` and upload data when `uploaded_generation_ != generation_`.